src/Tribe/Tickets_View.php

```diff
@@ -1080,6 +1080,13 @@ public function get_rsvp_block( $post, $echo = true ) {
 		/** @var Tribe__Tickets__RSVP $rsvp */
 		$rsvp = tribe( 'tickets.rsvp' );
 
+		// Check if the call is coming from a shortcode.
+		$doing_shortcode = tribe_is_truthy( array_search( 'do_shortcode', array_column( debug_backtrace(), 'function' ) ) );
```

Comment:
:warning: debug_backtrace() found. Debug code should not normally be used in production.

<sub>hash: 3bfee8bd910c9851aa9046f0f2a78aa5</sub>

Reply:
@sc0ttkclark do you have any idea on how we could be more tidy about detecting if it's doing shortcode?

Reply:
We could hook into `pre_do_shortcode_tag` to mark that we are "doing" a specific shortcode. Then hook into `do_shortcode_tag` to mark that we are done with that shortcode.

Maybe something for Tribe Common perhaps?

Reply:
How would that be? I'm not sure if I'm understanding. 

This is serving the purpose perfectly way. It's a perfect way of checking if it's being called from the shortcode or not. I feel it's something we should also have for the tickets block and the other shortcodes (Some people asked about it on the forums, so they can do different styling or even hiding if it's not coming from shortcode).

Reply:
Too bad we don't have this https://core.trac.wordpress.org/ticket/47616

I will lean on judgement from @lucatume on whether this is acceptable use of debug_backtrace() on production code.

Reply:
`debug_backtrace` is computationally very intensive and should not be used in production code, I would use a different approach.
@sc0ttkclark suggestion to hook on the shortcode filter to detect if we're in shortcode context or not is the way I would take. Ideally make that something that lives in `common` to be able to reuse it across plugins, the code provided in the patch linked by Scott is a good starting point for this.

Reply:
Cool, thanks! here's an implementation for that https://github.com/moderntribe/tribe-common/pull/1437

Reply:
✅